migration-helpers/update-api-folder-structure.js

```diff
@@ -0,0 +1,254 @@
+ * @param {string} apiName Name of the API
+ */
+const updateRoutes = async (apiPath, apiName) => {
+  const v3RoutePath = join(apiPath, "config", "routes.json");
+  const v4RoutePath = join(apiPath, "routes", `${apiName}.js`);
+  try {
+    // Create the js file
+    await fs.ensureFile(v4RoutePath);
+
+    // Create write stream for new js file
+    const file = fs.createWriteStream(v4RoutePath);
+    // Get the existing JSON routes file
+    const routesJson = await fs.readJson(v3RoutePath);
+    const { routes } = routesJson;
+
+    // Remove count
+    const updatedRoutes = routes.filter(
+      (route) =>
+        !route.handler.includes("count") || !route.path.includes("count")
+    );
+
+    // Transform objects to strings
+    const routesToString = inspect(
+      { routes: updatedRoutes },
+      { depth: Infinity }
+    );
+
+    // Export routes from create js file
+    file.write(`module.exports = ${routesToString}`);
+
+    // Close the write stream
+    file.end();
+
+    // Delete the v3 config/routes.json
+    await fs.remove(join(apiPath, "config", "routes.json"));
+  } catch (error) {
+    console.error(
+      `error: an error occured when migrating routes from ${v3RoutePath} to ${v4RoutePath}`
+    );
+  }
+};
+
+/**
+ *
+ * @param {string} apiPath Path to the current api
+ */
+const updatePolicies = async (apiPath) => {
+  const v3PoliciesPath = join(apiPath, "config", "policies");
+
+  const exists = await fs.exists(v3PoliciesPath);
+  if (!exists) return;
+
+  const v3Policies = await fs.readdir(v3PoliciesPath, { withFileTypes: true });
+  const policyFiles = v3Policies.filter((fd) => fd.isFile());
+
+  if (!policyFiles.length) {
+    await fs.remove(v3PoliciesPath);
+  }
+
+  const v4PoliciesPath = join(apiPath, "policies");
+  try {
+    for (const policy of policyFiles) {
+      await fs.copy(
+        join(v3PoliciesPath, policy.name),
+        join(v4PoliciesPath, policy.name)
+      );
+    }
+    // delete the v3 policy folder
+    await fs.remove(v3PoliciesPath);
+  } catch (error) {
+    console.error(
+      `error: an error occured when migrating a policy from ${v3PoliciesPath} to ${v4PoliciesPath}`
+    );
+  }
+};
+
+/**
+ *
+ * @description Recursively cleans a directory
+ *
+ * @param {array} dirs Directory entries
+ * @param {string} baseDir The path to the directory to clean
+ */
+const clean = async (dirs, baseDir) => {
+  for (dir of dirs) {
+    try {
+      const currentDirPath = join(baseDir, dir.name);
+      const currentDirContent = await fs.readdir(currentDirPath);
+
+      if (!currentDirContent.length) {
+        // Remove empty directory
+        await fs.remove(currentDirPath);
+      } else {
+        // Otherwise get the directories of the current directory
+        const currentDirs = await getDirsAtPath(currentDirPath);
+        await clean(currentDirs, currentDirPath);
+      }
+    } catch (error) {
+      console.error("error: failed to remove empty directories");
+    }
+  }
+};
+
```

Comment:
remove this and use `fs.rmdir(path, { recursive: true })` 🤦‍♂️

Reply:
Yes, I agree that we may not need to implement all of this ourselves.

Based on the `fs` docs, it looks like using `rmdir` with the recursive flag is deprecated, they recommend to use `fs.rm(path, { recursive: true })` instead

Reply:
I'm actually having second thoughts on this because this function recursively searches for _empty_ directories and removes them.  I think `fs.rm(path, { recursive: true })` will simply remove a directory and all its contents.  Maybe the function name isn't clear.